src/main/java/com/powsybl/openloadflow/ac/nr/NewtonRaphson.java

```diff
@@ -64,7 +66,36 @@ public static List<Pair<Equation<AcVariableType, AcEquationType>, Double>> findL
                 .collect(Collectors.toList());
     }
 
-    private NewtonRaphsonStatus runIteration(StateVectorScaling svScaling) {
+    public static List<Pair<Equation<AcVariableType, AcEquationType>, Double>> findLargestMismatch(EquationSystem<AcVariableType, AcEquationType> equationSystem, double[] mismatch, AcEquationType equationType) {
+        return equationSystem.getIndex().getSortedEquationsToSolve().stream()
+                .map(equation -> Pair.of(equation, mismatch[equation.getColumn()]))
+                .filter(e -> Math.abs(e.getValue()) > Math.pow(10, -7))
+                .filter(e -> e.getKey().getType() == equationType)
+                .sorted(Comparator.comparingDouble((Map.Entry<Equation<AcVariableType, AcEquationType>, Double> e) -> Math.abs(e.getValue())).reversed())
+                .limit(1)
+                .collect(Collectors.toList());
+    }
+
+    public static void reportLargestMismatch(Reporter reporter, EquationSystem<AcVariableType, AcEquationType> equationSystem, double[] mismatch, LfNetwork network, int iteration) {
+        for (AcEquationType acEquationType : List.of(AcEquationType.BUS_TARGET_P, AcEquationType.BUS_TARGET_Q, AcEquationType.BUS_TARGET_V)) {
+            findLargestMismatch(equationSystem, mismatch, acEquationType)
```

Comment:
here calling 3 times `findLargestMismatch`, hence hitting all equations 3 times, probably a performance hit.

also separating the 3 equation types in 3 reports is not very convenient. put them all in single report.